src/main/java/seedu/mark/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,153 @@
+package seedu.mark.logic.commands;
+
+import static seedu.mark.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import seedu.mark.commons.core.LogsCenter;
+import seedu.mark.commons.exceptions.DataConversionException;
+import seedu.mark.logic.commands.commandresult.CommandResult;
+import seedu.mark.logic.commands.exceptions.CommandException;
+import seedu.mark.model.Model;
+import seedu.mark.model.ReadOnlyMark;
+import seedu.mark.model.bookmark.Bookmark;
+import seedu.mark.storage.JsonMarkStorage;
+import seedu.mark.storage.Storage;
+
+/**
+ * Imports Mark's {@code Bookmark} data from a specific file.
+ */
+public class ImportCommand extends Command {
+
+    public static final String COMMAND_WORD = "import";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports all of Mark's data from the specified file.\n"
+            + "Parameters: FILENAME\n"
+            + "Example: " + COMMAND_WORD + " savedBookmarks";
+
+    public static final String MESSAGE_IMPORT_SUCCESS = "Bookmarks successfully imported from %1$s";
+    public static final String MESSAGE_IMPORT_SUCCESS_WITH_DUPLICATES = MESSAGE_IMPORT_SUCCESS + "\n\n"
+            + "The following bookmarks were not imported as they already exist: %2$s";
+    public static final String MESSAGE_IMPORT_FAILURE = "There was a problem while trying to import bookmarks "
+            + "from this file";
+    public static final String MESSAGE_FILE_FORMAT_INCORRECT = "The format of data in the file %1$s is incorrect";
+    public static final String MESSAGE_FILE_NOT_FOUND = "The file %1$s does not exist";
+
+    private static final Logger logger = LogsCenter.getLogger(JsonMarkStorage.class);
+
+    private final Path filePath;
+
+    /**
+     * Creates a new {@code ImportCommand} instance with the given file path.
+     *
+     * @param filePath of the file to be imported from.
+     */
+    public ImportCommand(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Converts a given list of {@code Bookmark}s to a multi-line {@code String},
+     * with each line having an indent of four spaces.
+     *
+     * @param bookmarks List of bookmarks.
+     * @return String representation of the bookmark list with indentation.
+     */
+    public static String toIndentedString(List<Bookmark> bookmarks) {
```

Comment:
I think this method can be extracted somewhere else, doesn't feel like it belongs here?